index.js

```diff
@@ -227,4 +227,35 @@ transform.resetTypes = function resetTypes() {
   registerTypes = {};
 };
 
-module.exports = transform;
+function addFieldPermissions(permission, fields, allKeys, template) {
+    Object.entries(fields).forEach((data) => {
+    const [key, value] = data
+    let exists = []
+    Object.entries(value.fields).forEach((subData) => {
+        const [subKey, subValue] = subData
+        if (allKeys.includes(subKey)) {
+          exists.push(subKey)
+        }
+        if (!subValue["ui:component"] || subValue["ui:component"].includes('auto-fill')) {
```

Comment:
When the `ui:component` is not set or it has `auto-fill` then template for this field, by default, is a textbox from. And the default textbox can't be overridden, it can only be cloned and in the cloned textbox we cannot pass the editable and viewable flag.

Reply:
I think @pseudorandumb comment on the other PR would take care of this?
> tcomb-json-schema shouldn't know about the CustomTextArea component (Textbox) here. transform.registerComponent('textarea', Textarea) associates 'textarea' in the ui_schema to the component.

Does that answer this question @vlade-symphony ?

If not, @pseudorandumb, do you know how to address the issue Vlad is seeing?
> the default textbox can't be overridden, it can only be cloned and in the cloned textbox we cannot pass the editable and viewable flag.

Reply:
You can't currently override the base types in this shared transform code, it currently explicitly rejects it. We need to change the code so that if a base type is passed in, use the given Factory/Type, otherwise use the default. Then yes a Factory can be created and associated with t.String and used with `transform.registerComponent('textarea', Textarea)`, similar to how textarea.jsx is used for the custom text box.